src/cast_core/include/cast_core/actors/OptionsConversion.hpp

```diff
@@ -118,37 +125,277 @@ struct NodeConvert<mongocxx::options::count> {
+            arr.append(toArrayBson(node));
+            return;
+        case YAML::NodeType::Undefined:
+        case YAML::NodeType::Scalar:
+        case YAML::NodeType::Null: {
+            Value val = Value::parseScalar(node);
+            val.appendToBuilder(arr);
+            return;
+        }
+    }
+
+    std::abort();
+}
+
+bsoncxx::document::value toDocumentBson(const YAML::Node& node) {
+    if (!node.IsMap()) {
+        std::stringstream msg;
+        msg << "Wanted map got " << node.Type() << ": " << to_string(node);
+        BOOST_THROW_EXCEPTION(InvalidYAMLToBsonException(msg.str()));
+    }
+
+    bsoncxx::builder::basic::document doc{};
+    for (auto&& kvp : node) {
+        appendToBuilder(kvp.second, kvp.first.as<std::string>(), doc);
+    }
+    return doc.extract();
+}
+
+bsoncxx::array::value toArrayBson(const YAML::Node& node) {
+    if (!node.IsSequence()) {
+        std::stringstream msg;
+        msg << "Wanted sequence got " << node.Type() << ": " << to_string(node);
+        BOOST_THROW_EXCEPTION(InvalidYAMLToBsonException(msg.str()));
+    }
+
+    bsoncxx::builder::basic::array arr{};
+    for (auto&& elt : node) {
+        appendToBuilder(elt, arr);
+    }
+    return arr.extract();
+}
+
+bsoncxx::document::value toDocumentBson(const genny::Node& node) {
+    std::stringstream str;
+    str << node;
+    const YAML::Node asYaml = YAML::Load(str.str());
+    return genny::toDocumentBson(asYaml);
+}
+
+bsoncxx::document::value toDocumentBson(const std::string& yaml) {
+    const YAML::Node asYaml = YAML::Load(yaml);
+    return genny::toDocumentBson(asYaml);
+}
+
+bsoncxx::array::value toArrayBson(const std::string& yaml) {
+    const YAML::Node asYaml = YAML::Load(yaml);
+    return genny::toArrayBson(asYaml);
+}
+
+}  // namespace
+
+template <>
+struct NodeConvert<bsoncxx::document::value> {
+    using type = bsoncxx::document::value;
+
+    static type convert(const Node& node) {
+        return toDocumentBson(node);
+    }
+};
+
 template <>
 struct NodeConvert<mongocxx::options::find> {
     using type = mongocxx::options::find;
 
     static type convert(const Node& node) {
         type rhs{};
 
-        if (node["Hint"]) {
-            auto h = node["Hint"].to<std::string>();
-            auto hint = mongocxx::hint(std::move(h));
-            rhs.hint(mongocxx::hint(hint));
+        if (const auto& sort = node["Sort"]) {
+            rhs.sort(sort.to<bsoncxx::document::value>());
         }
-        if (node["Comment"]) {
-            auto c = node["Comment"].to<std::string>();
-            rhs.comment(std::move(c));
+        if (const auto& collation = node["Collation"]) {
+            rhs.collation(collation.to<bsoncxx::document::value>());
         }
-        if (node["Limit"]) {
-            auto limit = node["Limit"].to<int>();
-            rhs.limit(limit);
+        if (const auto& hint = node["Hint"]) {
+            rhs.hint(mongocxx::hint(std::move(hint.to<std::string>())));
```

Comment:
The fact that we only support string hints isn't ideal, but I guess it works ok for now. Perhaps we should at least leave a comment here stating that in the future this could be extended to work with hints expressed either as strings or documents.